src/App.tsx

```diff
@@ -1,61 +1,105 @@
-import React from 'react';
-import 'bulma/css/bulma.css';
-import './App.scss';
+import classNames from "classnames";
+import React, { useState } from "react";
+import "bulma/css/bulma.css";
+import "./App.scss";
 
 export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
+  "Dumplings",
+  "Carrot",
+  "Eggs",
+  "Ice cream",
+  "Apple",
+  "Bread",
+  "Fish",
+  "Honey",
+  "Jam",
+  "Garlic",
 ];
 
+export function getReorderedGoods(
+  goods: string[],
+  sortType: string,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((itemOne: string, itemTwo: string): number => {
+    switch (sortType) {
+      case "ALPHABET":
+        return itemOne.localeCompare(itemTwo);
+      case "LENGTH":
+        return itemOne.length - itemTwo.length;
+      default:
+        return 0;
+    }
+  });
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState("NONE");
```

Comment:
Can't see 'NONE' as a case at your switch statement. 
In this case, it's better to pass `undefined` as a initial value if the idea is to trigger default option at switch.